### PR TITLE
use matroska.py and file2string.py as python modules

### PR DESCRIPTION
**Pull request by pigoz**

### CODE REVIEW

File: waftools/generators/sources.py [hunk 1]
```diff
@@ -21,20 +20,25 @@ def __file2string__(ctx, **kwargs):
...
+
+@TaskGen.feature('ebml_definitions')
+def ebml_definitions(self):
+    setattr(self, 'rule', ' ')
+    target = getattr(self, 'target', None)
+    out = self.path.find_or_declare(target)
+    tmp = StringIO()
+    generate_C_definitions(tmp)
+    out.write(tmp.getvalue())
+    tmp.close()
```
Comment (wm4) [inline waftools/generators/sources.py]: Would it be possible to factorize some of this code? it's repeated 3 times with file2string.

Comment (pigoz) [inline waftools/generators/sources.py]: refactored in the following commit

**Comment by wiiaboo** [comment]
In case it was missed on IRC, I tested this with both py2 and py3 and generated files were identical between them.

**Comment by pigoz** [comment]
@wiiaboo great!

So, should I merge? I believe @wm4 didn't like the move of matroska.py and file2string.py from tools to waftools.

**Comment by wm4** [comment]
> I believe @wm4 didn't like the move of matroska.py and file2string.py from tools to waftools.

I don't like that matroska.py is moved, other than that look good to me. Maybe you could just make it a symlink from waftools to TOOLS matroska.py?

**Comment by pigoz** [comment]
@wm4 you mean a symlink in the actual committed source tree? will that be ok when [cross-]compiling on windows?

**Comment by wm4** [comment]
Yes, I think git will make a copy.

**Comment by pigoz** [comment]
changed to use symlinks as you suggested

**Comment by wm4** [comment]
OK

**Comment by wiiaboo** [comment]
symlinks don't work on Windows. Git only creates a blob with the path to the linked file. You could switch the place of the files by moving them to waftools and symlinking them to TOOLS instead.

**Comment by wm4** [comment]
I was relatively sure it'd create a copy, but maybe it depends on git client and settings.

Can't it somehow import the file directly from the TOOLS path? Just opening a python module by path seems like a PITA, but it looks like just adding another loader path seems to be simple, as outlined in the second answer here: http://stackoverflow.com/questions/67631/how-to-import-a-module-given-the-full-path

**Comment by pigoz** [comment]
Yeah I'll add `TOOLS` to the load path like we do for `waftools` here. https://github.com/mpv-player/mpv/blob/master/wscript#L4

**Comment by pigoz** [comment]
rebased and merged 737e3b1

**Pull request closed without merging**